src/Orleans.Core/Configuration/Options/GrainCollectionOptions.cs

```diff
@@ -8,12 +9,47 @@ namespace Orleans.Hosting
     /// </summary>
     public class GrainCollectionOptions
     {
+        /// <summary>
+        /// Regulates the periodic collection of inactive grains.
+        /// </summary>
         public TimeSpan CollectionQuantum { get; set; } = DEFAULT_COLLECTION_QUANTUM;
         public static readonly TimeSpan DEFAULT_COLLECTION_QUANTUM = TimeSpan.FromMinutes(1);
 
+        /// <summary>
+        /// Default period of inactivity necessary for a grain to be available for collection and deactivation.
+        /// </summary>
         public TimeSpan CollectionAge { get; set; } = DEFAULT_COLLECTION_AGE_LIMIT;
         public static readonly TimeSpan DEFAULT_COLLECTION_AGE_LIMIT = TimeSpan.FromHours(2);
 
+        /// <summary>
+        /// Period of inactivity necessary for a grain to be available for collection and deactivation by grain type.
+        /// </summary>
         public Dictionary<string, TimeSpan> ClassSpecificCollectionAge { get; set; } = new Dictionary<string, TimeSpan>();
     }
+
+    public class GrainCollectionOptionsFormatter : IOptionFormatter<GrainCollectionOptions>
+    {
+        public string Category { get; }
+
+        public string Name => nameof(GrainCollectionOptions);
+        private GrainCollectionOptions options;
+        public GrainCollectionOptionsFormatter(IOptions<GrainCollectionOptions> options)
+        {
+            this.options = options.Value;
+        }
+
+        public IEnumerable<string> Format()
+        {
+            var formated = new List<string>()
+            {
+                OptionFormattingUtilities.Format(nameof(this.options.CollectionQuantum), this.options.CollectionQuantum),
+                OptionFormattingUtilities.Format(nameof(this.options.CollectionAge), this.options.CollectionAge),
+            };
+            foreach(KeyValuePair<string, TimeSpan> classCollectionAge in this.options.ClassSpecificCollectionAge)
```

Comment:
Do we need a null check before enumerating this?

Reply:
Null ClassSpecificCollectionAge is not valid.  Think we should probably handle that in a validator if/when necessary.